### PR TITLE
feat(account-store): made the account store

### PR DESCRIPTION
**Pull request by PedroFonseca17**
Made the account store and the API calls for account details, request watchlist, and rated movies.

### CODE REVIEW

File: src/store/account/thunks.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { IMovie } from '../../api/models';
+
+export const fetchAccountDetails = (
+  sessionId: string
+): AppThunk => async dispatch => {
+  dispatch(requestAccountDetailsStart());
+  getAccountDetails(sessionId)
+    .then(response => {
+      const account = response.data as IAccount;
+      console.log('account', account);
```
Comment (FilipeAVS) [inline src/store/account/thunks.ts]: Remove this log here.

File: src/store/account/reducers.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      isFetchingRatedMovies: true,
+      failedFetchingRatedMovies: false,
+      watchList: undefined
+    };
+  case REQUEST_RATED_MOVIES_SUCCESSS:
+    return {
+      ...state,
+      isFetchingRatedMovies: false,
+      failedFetchingRatedMovies: false,
+      watchList: action.ratedMovies
```
Comment (FilipeAVS) [inline src/store/account/reducers.ts]: I think this should be `ratedMovies: ...`

File: src/api/api.ts [hunk 3]
```diff
@@ -49,14 +49,26 @@ export const createRequestToken = () =>
...
+export const getAccountDetails = (sessionId: string) =>
+  exec(`account?session_id=${sessionId}`, Method.GET, {
+    session_id: sessionId
+  });
+
+export const getRatedMovies = (accontId: number) =>
+  exec(`account/${accontId}/rated/movies`, Method.GET);
+
+export const getWatchList = (accontId: number) =>
+  exec(`account/${accontId}/watchlist/movies`, Method.GET);
```
Comment (tiagop93) [inline src/api/api.ts]: TYPO: accontId

**Review by FilipeAVS — CHANGES_REQUESTED** [review]
A couple of changes requested.

Also, a thought: We could store the account and the session ID on the same store, as they are connected.

**Review by tiagop93 — COMMENTED** [review]
Left a comment.

**Review by FilipeAVS — APPROVED** [review]

**Pull request merged**